core/src/mindustry/editor/MapLocalesDialog.java

```diff
@@ -0,0 +1,759 @@
+package mindustry.editor;
+
+import arc.Core;
+import arc.func.*;
+import arc.graphics.*;
+import arc.scene.style.*;
+import arc.scene.ui.*;
+import arc.scene.ui.layout.*;
+import arc.scene.utils.*;
+import arc.struct.*;
+import arc.util.*;
+import mindustry.*;
+import mindustry.ctype.*;
+import mindustry.game.*;
+import mindustry.gen.*;
+import mindustry.graphics.*;
+import mindustry.io.*;
+import mindustry.type.*;
+import mindustry.ui.*;
+import mindustry.ui.dialogs.*;
+
+import static mindustry.Vars.*;
+
+public class MapLocalesDialog extends BaseDialog{
```

Comment:
There's way too much code here for me to review all at once, but here's an issue I ran into immediately: Why isn't there English by default?

When I open the dialog, there are no languages on the left. I add an entry:

![Screenshot from 2023-11-09 10-19-00](https://github.com/Anuken/Mindustry/assets/10100000/0b1679c5-9abd-428d-960a-b324e4fcb14d)

If I then proceed to add English, the entry disappears. Where did it go? I can't select it or go back to it:

![Screenshot from 2023-11-09 10-19-18](https://github.com/Anuken/Mindustry/assets/10100000/ed914f9b-2b3e-4e12-a1ed-7d9e2c82c3a4)

When I copy the text, it shows up like this:

```
default:
name = text
en:
```

Why is there a "default" and an "en" section? Shouldn't en be the default? If not, why isn't there a "default" entry on the left?

Reply:
"default" locale just didn't display on the left as it isn't presented in Vars.locales and things like language dialog. It should've been replaced by Locale.getDefault().getLanguage(), which is done now. The property you've added just left in this default locale (which you couldn't switch to) as they don't copy when you add a new locale bundle, there are separate edit buttons for doing so.